libs/balances-and-allowances/src/updaters/BalancesCacheUpdater.tsx

```diff
@@ -14,11 +14,7 @@ interface BalancesCacheUpdaterProps {
 export function BalancesCacheUpdater({ chainId, account }: BalancesCacheUpdaterProps): null {
   const [balances, setBalances] = useAtom(balancesAtom)
   const [balancesCache, setBalancesCache] = useAtom(balancesCacheAtom)
-  const areBalancesRestoredFromCacheRef = useRef(false)
-
-  useEffect(() => {
-    areBalancesRestoredFromCacheRef.current = false
```

Comment:
Simplified logic by assigning chainId as value to the ref, so we don't need to reset the value when chainId changes